AutoMapper.AspNetCore.OData.EF6/QueryableExtensions.cs

```diff
@@ -50,22 +50,9 @@ public static class QueryableExtensions
         /// <param name="mapper"></param>
         /// <param name="options"></param>
         /// <param name="handleNullPropagation"></param>
-        /// <returns></returns>
-        public static async Task<ICollection<TModel>> GetAsync<TModel, TData>(this IQueryable<TData> query, IMapper mapper, ODataQueryOptions<TModel> options, HandleNullPropagationOption handleNullPropagation = HandleNullPropagationOption.Default)
-            where TModel : class
-            => await query.GetAsync(mapper, options, new QuerySettings { ODataSettings = new ODataSettings { HandleNullPropagation = handleNullPropagation } });
-
-        /// <summary>
-        /// GetAsync
-        /// </summary>
-        /// <typeparam name="TModel"></typeparam>
-        /// <typeparam name="TData"></typeparam>
-        /// <param name="query"></param>
-        /// <param name="mapper"></param>
-        /// <param name="options"></param>
         /// <param name="querySettings"></param>
         /// <returns></returns>
-        public static async Task<ICollection<TModel>> GetAsync<TModel, TData>(this IQueryable<TData> query, IMapper mapper, ODataQueryOptions<TModel> options, QuerySettings querySettings = null)
+        public static async Task<ICollection<TModel>> GetAsync<TModel, TData>(this IQueryable<TData> query, IMapper mapper, ODataQueryOptions<TModel> options, HandleNullPropagationOption handleNullPropagation = HandleNullPropagationOption.Default, QuerySettings querySettings = null)
```

Comment:
In this case `handleNullPropagation` is not used at all. What was the problem with the previous signatures? I'm not sure we should have both `handleNullPropagation` and `querySettings` parameters in a single method. It might add confusion since `querySettings` has `HandleNullPropagation` property as well.

Reply:
You're correct that it is confusing but changing public signatures breaks calling code.  We should add the `[Obsolete]` attribute and give `handleNullPropagation` precedence for the `ToFilterExpression` argument only - the way it was before.  This signature will be removed (in favor of the one below it) in the next major release as will all methods with the `HandleNullPropagationOption handleNullPropagation` parameter.

Reply:
oh that's why I was a bit confused with your change, after everything is done I'll squash all commits into a single one, to make history clean